src/fees.ts

```diff
@@ -28,76 +30,141 @@ const MAX_FEE_RATE_SATS_PER_VBYTE = new BigNumber(1000); // 1000 Sats/vbyte
 const MAX_FEE_SATS = new BigNumber(2500000); // ~ 0.025 BTC ~ $250 if 1 BTC = $10k
 
 /**
- * Validate the given transaction fee rate (in Satoshis/vbyte).
- *
- * - Must be a parseable as a number.
- *
- * - Cannot be negative (zero is OK).
- *
- * - Cannot be greater than the limit set by
- *   `MAX_FEE_RATE_SATS_PER_VBYTE`.
+ * Provide a readable message from a FeeValidationError for user display.
  */
-export function validateFeeRate(feeRateSatsPerVbyte) {
-  let fr;
-  try {
-    fr = new BigNumber(feeRateSatsPerVbyte);
-  } catch (e) {
-    return "Invalid fee rate.";
-  }
-  if (!fr.isFinite()) {
-    return "Invalid fee rate.";
-  }
-  if (fr.isLessThan(ZERO)) {
-    return "Fee rate cannot be negative.";
-  }
-  if (fr.isGreaterThan(MAX_FEE_RATE_SATS_PER_VBYTE)) {
-    return "Fee rate is too high.";
+export function getFeeErrorMessage(error: FeeValidationError | null) {
+  let errorMessage = "";
+
+  switch (error) {
+    case FeeValidationError.FEE_CANNOT_BE_NEGATIVE:
+      errorMessage = "Fee cannot be negative.";
```

Comment:
Can't these just return the string at the end of each case?

Reply:
They could be, but I don't like using returns in switch statements. There is already a conventional command to interrupt switch statement fall-through. Introducing a different one isn't needed and may result in a bigger change in the future if modifying the switch statement or function. If this single exit-point pattern isn't acceptable, I'd probably change the function to use if statements, but I think the switch is more performant when checking this type.

Reply:
could I convince you to use an object mapping here then?
```ts
const feeErrorMessages = {
  [FeeValidationError.FEE_CANNOT_BE_NEGATIVE]: "Fee cannot be negative.",
  [FeeValidationError.FEE_RATE_CANNOT_BE_NEGATIVE]: "Fee rate cannot be negative.",
  // ... other mappings
};

export function getFeeErrorMessage(error: FeeValidationError | null): string {
  return error !== null ? feeErrorMessages[error] ?? "" : "";
}
```

Reply:
oh nevermind lol

Reply:
whoops. sorry.

Reply:
Nothing released yet so we can revert

Reply:
The mapping is nice.

Reply:
Six of one, a half dozen of the other. It's all the same to me, really.